### ISSUE TITLE
How to use your own models in this notebook

### ISSUE THREAD

**Issue by mindblowingaiart** [dobrosketchkun/latent_space_adventures#1]
Hi I wanted to try out mixing the layers of two of my model .pkls using your notebook, I can get it to work for the models which come in the notebook (when you run the code line and it downloads them), but if I mount my drive and move MY .pkls to the same spot as the ones installed automatically by the "neural networks handling" cell, and try to blend those networks together it doesn't work? Not sure what I am doing wrong...

When I go to insert text here for the mix layers step - if I just put the name of the .pkl it doesn't work, unless I use the networks which were downloaded in the "neural networks handling" step...In which case it works if I just put in two of the names from the downloaded ones.

If I use my own networks for layers do I need to run the 'utility function step' for it to work or just "installs" & "mix layer"?

Hopefully there is an easy fix for this or something simple I am missing ( I am not a coder so just looking for an easy explanation how I could do this if possible). This seems to be the easiest notebook I have found for network blending that I can wrap my head around, now if I could just get it to work for my own networks... thanks for making this.

**Comment by dobrosketchkun** [comment]
>if I mount my drive and move MY .pkls to the same spot as the ones installed automatically by the "neural networks handling" cell, and try to blend those networks together it doesn't work? Not sure what I am doing wrong...   

So you are saying you mount your google drive and copy your styleGAN2-ada model .pkl files in /content/ folder? What type of error are we talking about?

>If I use my own networks for layers do I need to run the 'utility function step' for it to work 

Yes, you should run it.

**Comment by mindblowingaiart** [comment]
let me give it another go - ill try to provide you some more information

**Comment by mindblowingaiart** [comment]
So basically all I am asking is - 
1)Say these two pickles - bear.pkl and lion.pkl, what cells do i need to run in order to get the layer mixing to work, and how do I input these .pkls into the code to make it work? Would I do this at the utility function step or the mix layers? 

2)Is it crucial I mount my drive before I run any of the other cells if I plan to mix layers from my own networks?

I actually know you can click "show code and I can input it that way. Sorry if these questions seem stupid, I am not a coder, I am just trying to figure out this specific functionality of this notebook, thanks so much for your help

**Comment by mindblowingaiart** [comment]
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-4-6745f122b0df> in <module>()
     38 network_pkl0 = '/content/' + network_name2.strip() + '.pkl'
     39 
---> 40 with dnnlib.util.open_url(network_pkl0) as f:
     41     G = legacy.load_network_pkl(f)['G_ema'].to(device) # type: ignore
     42 

/content/stylegan2-ada-pytorch/dnnlib/util.py in open_url(url, cache_dir, num_attempts, verbose, return_filename, cache)
    387     # Doesn't look like an URL scheme so interpret it as a local filename.
    388     if not re.match('^[a-z]+://', url):
--> 389         return url if return_filename else open(url, "rb")
    390 
    391     # Handle file URLs.  This code handles unusual file:// patterns that

FileNotFoundError: [Errno 2] No such file or directory: '/content//content/drive/MyDrive/Colab Notebooks/good pickles/upphq portraits w autocrop 1.pkl.pkl'

**Comment by mindblowingaiart** [comment]
I figured out what to do -  
1)mount google drive

from google.colab import drive
drive.mount('/content/drive')

2)I had to replace your .json link with my .json link, and then I uploaded the .json file itself into the content folder of Gdrive. In order to run the cell properly first you have to click the 3 verticle dots on the right of the "nueral networks handling cell", click show code, and then change the .json link to your github link (which you have to create beforehand by making a repository on github and uploading the .json file to gdrive content folder). 

3)Next change the code right below this to the actual .json file location itself - see where it says 'content/...' right under the github 'raw' .json url link you just changed. 

Thats what you will replace with the file path of the .json file(you will need to upload the .json file to your gdrive content folder and click copy path). the .json file structure looks like this: 
{"portraits" : "https://drive.google.com/file/d/10hCoR57tc0q40_WwezxSOAiNXKmqPE/view?usp=sharing", 
"variety" : "https://drive.google.com/file/d/10hCoR57tc0q40_WyGweSOAiNXKmqPE/view?usp=sharing"} 

After the cell runs, you will be able to type in the names for each network which you want to mix.

**Comment by dobrosketchkun** [comment]
Yeah, it's one way to address this issue. I'm glad you figured it out since I'm was kind of slow on a solution.
